src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.FullGlobalizationData.cs

```diff
@@ -7,260 +7,75 @@ namespace System
-            "Pacific/Pitcairn"    // Prefer "Pitcairn Islands Time" over "Pitcairn Time"
-        };
-
-        // Main function that is called during construction to populate the three display names
-        private static void TryPopulateTimeZoneDisplayNamesFromGlobalizationData(string timeZoneId, TimeSpan baseUtcOffset, ref string? standardDisplayName, ref string? daylightDisplayName, ref string? displayName)
-        {
-            // Determine the culture to use
-            CultureInfo uiCulture = CultureInfo.CurrentUICulture;
-            if (uiCulture.Name.Length == 0)
-                uiCulture = CultureInfo.GetCultureInfo(FallbackCultureName); // ICU doesn't work nicely with InvariantCulture
-
-            // Attempt to populate the fields backing the StandardName, DaylightName, and DisplayName from globalization data.
-            GetDisplayName(timeZoneId, Interop.Globalization.TimeZoneDisplayNameType.Standard, uiCulture.Name, ref standardDisplayName);
-            GetDisplayName(timeZoneId, Interop.Globalization.TimeZoneDisplayNameType.DaylightSavings, uiCulture.Name, ref daylightDisplayName);
-            GetFullValueForDisplayNameField(timeZoneId, baseUtcOffset, uiCulture, ref displayName);
-        }
-
-        // Helper function to get the standard display name for the UTC static time zone instance
-        private static string GetUtcStandardDisplayName()
-        {
-            // Don't bother looking up the name for invariant or English cultures
-            CultureInfo uiCulture = CultureInfo.CurrentUICulture;
-            if (GlobalizationMode.Invariant || uiCulture.Name.Length == 0 || uiCulture.TwoLetterISOLanguageName == "en")
-                return InvariantUtcStandardDisplayName;
-
-            // Try to get a localized version of "Coordinated Universal Time" from the globalization data
-            string? standardDisplayName = null;
-            GetDisplayName(UtcId, Interop.Globalization.TimeZoneDisplayNameType.Standard, uiCulture.Name, ref standardDisplayName);
-
-            // Final safety check.  Don't allow null or abbreviations
-            if (standardDisplayName == null || standardDisplayName == "GMT" || standardDisplayName == "UTC")
-                standardDisplayName = InvariantUtcStandardDisplayName;
-
-            return standardDisplayName;
-        }
-
-        // Helper function to get the full display name for the UTC static time zone instance
-        private static string GetUtcFullDisplayName(string timeZoneId, string standardDisplayName)
-        {
-            return $"(UTC) {standardDisplayName}";
-        }
-
-        // Helper function that retrieves various forms of time zone display names from ICU
-        private static unsafe void GetDisplayName(string timeZoneId, Interop.Globalization.TimeZoneDisplayNameType nameType, string uiCulture, ref string? displayName)
+        private static unsafe bool TryConvertIanaIdToWindowsId(string ianaId, bool allocate, out string? windowsId)
```

Comment:
Doesn't matter much because it is private but it helps with data flow analysis for nullability in code that calls this, you could use NotNullWhen(true).

Reply:
It can be null even when we return true here. Therefore I annotated the public API and not this one.

Reply:
A, ok I missed that.